src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LegacyBqlField/LegacyBqlFieldAnalyzer.cs

```diff
@@ -0,0 +1,78 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LegacyBqlField
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class LegacyBqlFieldAnalyzer : PXDiagnosticAnalyzer
+	{
+		public const string CorrespondingPropertyType = "CorrespondingPropertyType";
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Descriptors.PX1060_LegacyBqlField);
+
+		protected override bool ShouldAnalyze(PXContext pxContext) => pxContext.IsAcumatica2019R1;
+
+		internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+		{
+			compilationStartContext.RegisterSymbolAction(
+				c => Analyze(c, pxContext),
+				SymbolKind.NamedType);
+		}
+
+		private async void Analyze(SymbolAnalysisContext context, PXContext pxContext)
+		{
+			if (context.Symbol is INamedTypeSymbol dacFieldType)
+			{
+				if (!IsDacFieldType(dacFieldType, pxContext) || AlreadyStronglyTyped(dacFieldType, pxContext))
+					return;
+
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				Location location = dacFieldType.Locations.FirstOrDefault();
+				if (location != null)
+				{
+					var property = dacFieldType.ContainingType
+						.GetBaseTypesAndThis()
+						.SelectMany(t => t.GetMembers().OfType<IPropertySymbol>())
+						.FirstOrDefault(f => String.Equals(f.Name, dacFieldType.Name, StringComparison.OrdinalIgnoreCase));
+
+					if (property != null)
+					{
+						string propertyTypeName = property.Type is IArrayTypeSymbol arrType
+							? arrType.ElementType.Name + "[]"
+							: property.Type is INamedTypeSymbol namedType
+								? namedType.IsNullable(pxContext)
+									? namedType.GetUnderlyingTypeFromNullable(pxContext).Name
+									: namedType.Name
+								: null;
+
+						if (propertyTypeName == null || !LegacyBqlFieldFix.PropertyTypeToFieldType.ContainsKey(propertyTypeName))
+							return;
+
+						context.CancellationToken.ThrowIfCancellationRequested();
+
+						var properties = ImmutableDictionary.CreateBuilder<string, string>();
+						properties.Add(CorrespondingPropertyType, propertyTypeName);
+						context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1060_LegacyBqlField, location, properties.ToImmutable(), dacFieldType.Name));
+					}
+				}
+			}
+		}
+
+		private static bool IsDacFieldType(ITypeSymbol dacFieldType, PXContext pxContext)
+			=> dacFieldType.TypeKind == TypeKind.Class &&
+				dacFieldType.IsDacField() &&
+				dacFieldType.BaseType.SpecialType == SpecialType.System_Object &&
+				dacFieldType.ContainingType != null &&
+				dacFieldType.ContainingType.IsDacOrExtension(pxContext);
+
+		internal static bool AlreadyStronglyTyped(INamedTypeSymbol dacFieldType, PXContext pxContext)
+			=> dacFieldType.AllInterfaces.Any(t =>
+				t.IsGenericType
+				&& t.ConstructUnboundGenericType().Name == pxContext.IImplementType.Name
```

Comment:
@Derrekam why didn't you used `OriginalDefinition` instead of `ConstructUnboundGenericType()`

Reply:
Would it be more efficient to use ```OriginalDefinition```?

Reply:
Actually, yes. According to Roslyn reference source for C# `ConstructUnboundGenericType()` internally looks like this
```
 /// <summary>
 /// Returns an unbound generic type of this named type.
 /// </summary>
 public NamedTypeSymbol ConstructUnboundGenericType()
  {
       return OriginalDefinition.AsUnboundGenericType();
  }
```
The `AsUnboundGenericType` internals:
```
static public NamedTypeSymbol AsUnboundGenericType(this NamedTypeSymbol type)
{
	if (!type.IsGenericType)
	{
		// This exception is part of the public contract of NamedTypeSymbol.ConstructUnboundGenericType
		throw new InvalidOperationException();
	}

	var original = type.OriginalDefinition;
	int n = original.Arity;
	NamedTypeSymbol originalContainingType = original.ContainingType;

	var constructedFrom = ((object)originalContainingType == null) ?
		original :
		original.AsMember(originalContainingType.IsGenericType ? originalContainingType.AsUnboundGenericType() : originalContainingType);
	if (n == 0)
	{
		return constructedFrom;
	}

	var typeArguments = UnboundArgumentErrorTypeSymbol.CreateTypeArguments(
		constructedFrom.TypeParameters,
		n,
		new CSDiagnosticInfo(ErrorCode.ERR_UnexpectedUnboundGenericName));
	return constructedFrom.Construct(typeArguments, unbound: true);
}
```
As you can see there is a lot of additional actions for `ConstructUnboundGenericType`. We didn't encounter any problems with 'OriginalDefinition' so I think you can safely use it.